### PR TITLE
Add gpu kernel for new api : linalg.lstsq

### PR DESCRIPTION
**Pull request by haohongxiang**
### PR types

Function optimization
### PR changes

OPs
### Describe

Add gpu kernel for new api : linalg.lstsq
You can see the implementation of cpu kernel in [PR38585](https://github.com/PaddlePaddle/Paddle/pull/38585)
The docs_cn of paddle.linalg.lstsq is merged by [docs-PR4174](https://github.com/PaddlePaddle/docs/pull/4174)

1、usage example：
```
import paddle
paddle.set_device("cpu")
x = paddle.to_tensor([[1, 3], [3, 2], [5, 6.]])
y = paddle.to_tensor([[3, 4, 6], [5, 3, 4], [1, 2, 1.]])
results = paddle.linalg.lstsq(x, y, driver="gelsd")

# solution
print(results[0]) 
# [[ 0.78350395, -0.22165027, -0.62371236],
# [-0.11340097,  0.78866047,  1.14948535]]

# residuals
print(results[1])
# [19.81443405, 10.43814468, 30.56185532])

# rank
print(results[2])
# 2

# singular values
print(results[3])
# [9.03455734, 1.54167950]
```

2、docs_cn：

image

3、docs_en：

image

### CODE REVIEW

**Review by dingjiaweiww — DISMISSED** [review]

File: python/paddle/tensor/linalg.py [hunk 1]
```diff
@@ -2617,8 +2617,65 @@ def __check_input(x, UPLO):
...
+    """
+    Computes a solution to
+    the least squares problem of a system of linear equations.
+
+    Args:
+        x (Tensor): A tensor with shape ``(*, M, N)`` , the data type of the input Tensor ``x``
+            should be one of float32, float64.
+        y (Tensor): A tensor with shape ``(*, M, K)`` , the data type of the input Tensor ``y`` 
+            should be one of float32, float64.
+        rcond(float, optional): A float pointing number used to determine the effective rank of ``x``.
```
Comment (TCChenlong) [inline python/paddle/tensor/linalg.py]: 需要说明默认值。

File: python/paddle/tensor/linalg.py [hunk 2]
```diff
@@ -2617,8 +2617,65 @@ def __check_input(x, UPLO):
...
+        with shape ``(*)``, meaning the ranks of the matrices in ``x``, which is computed when ``driver`` in 
+        (‘gelsy’, ‘gelsd’, ‘gelss’), otherwise return an empty tensor. ``singular_values`` is a tensor with 
+        shape ``(*, min(M, N))``, meaning singular values of the matrices in ``x``, which is computed when 
+        ``driver`` in (‘gelsd’, ‘gelss’), otherwise return an empty tensor.
+
+    Examples:
+        .. code-block:: python
+
+            import paddle
+            import numpy as np
```
Comment (TCChenlong) [inline python/paddle/tensor/linalg.py]: 这行冗余 可以删除

Comment (haohongxiang) [reply python/paddle/tensor/linalg.py]: 已删

File: python/paddle/tensor/linalg.py (hunk 1 above)
Comment (haohongxiang) [reply python/paddle/tensor/linalg.py]: 已加

File: paddle/fluid/operators/lstsq_op.cu [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    auto tau_dims_vec = framework::vectorize<int>(x_dims);
+    tau_dims_vec.pop_back();
+    tau_dims_vec[tau_dims_vec.size() - 1] = min_mn;
+    Tensor tau = dito.Fill(tau_dims_vec, 0);
+    auto tau_data = tau.mutable_data<T>(context.GetPlace());
+
+    auto tmp_x = dito.Transpose(new_x);
+    auto tmp_y = dito.Transpose(new_y);
+    framework::TensorCopy(tmp_x, new_x.place(), &new_x);
+    framework::TensorCopy(tmp_y, new_y.place(), &new_y);
```
Comment (qingqing01) [inline paddle/fluid/operators/lstsq_op.cu]: Why need this copy?

File: paddle/fluid/operators/lstsq_op.cu [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+                                   x_data, x_stride, tau_data, tau_stride,
+                                   y_data, y_stride);
+
+    auto trans_r = dito.Transpose(new_x);
+    auto trans_b = dito.Transpose(new_y);
+    auto slice_r = dito.Slice(trans_r, {-2}, {0}, {min_mn});
+    auto slice_b = dito.Slice(trans_b, {-2}, {0}, {min_mn});
+    auto tmp_r = dito.TrilTriu(slice_r, 0, false);
+    framework::TensorCopy(tmp_r, new_x.place(), &new_x);
+    framework::TensorCopy(slice_b, new_y.place(), &new_y);
```
Comment (qingqing01) [inline paddle/fluid/operators/lstsq_op.cu]: same as above

File: paddle/fluid/operators/lstsq_op.cu (hunk 3 above)
Comment (haohongxiang) [reply paddle/fluid/operators/lstsq_op.cu]: Thank you for comments. It's not necessary so I removed it.

File: paddle/fluid/operators/lstsq_op.cu (hunk 4 above)
Comment (haohongxiang) [reply paddle/fluid/operators/lstsq_op.cu]: Thank you for comments. It's not necessary so I removed it.

**Review by TCChenlong — APPROVED** [review]
LGTM for API docs

**Review by ForFishes — APPROVED** [review]
LGTM

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**